### PR TITLE
refactor: error message in http api

### PR DESCRIPTION
**Pull request by baojinri**
# Which issue does this PR close?

Closes #

# Rationale for this change
 error messages in http api are not detailed enough
 

# What changes are included in this PR?
- modify response error message
- modify log error message

# Are there any user-facing changes?
none

# How does this change test
unit test & manual test

### CODE REVIEW

File: server/service/http/api.go [hunk 1]
```diff
@@ -150,15 +150,16 @@ func (a *API) getShardTables(writer http.ResponseWriter, req *http.Request) {
 	var getShardTables GetShardTables
 	err := json.NewDecoder(req.Body).Decode(&getShardTables)
 	if err != nil {
-		a.respondError(writer, ErrParseRequest, "decode request body failed")
+		log.Error("decode request body failed", zap.Error(err))
+		a.respondError(writer, ErrParseRequest, fmt.Sprintf("decode request body failed, cause: %v", err))
```
Comment (ZuLiangWang) [inline server/service/http/api.go:154]:
```suggestion
		a.respondError(writer, ErrParseRequest, fmt.Sprintf("decode request body failed, cause: %s", err.Error()))
```

File: server/service/http/api.go [hunk 2]
```diff
@@ -150,15 +150,16 @@ func (a *API) getShardTables(writer http.ResponseWriter, req *http.Request) {
...
 		return
 	}
 	log.Info("get shard tables request", zap.String("request", fmt.Sprintf("%+v", getShardTables)))
 
 	c, err := a.clusterManager.GetCluster(req.Context(), getShardTables.ClusterName)
 	if err != nil {
 		log.Error("get cluster failed", zap.String("clusterName", getShardTables.ClusterName), zap.Error(err))
-		a.respondError(writer, ErrGetCluster, fmt.Sprintf("get cluster failed, clusterName:%s", getShardTables.ClusterName))
+		a.respondError(writer, ErrGetCluster, fmt.Sprintf("get cluster failed, clusterName: %s, cause: %v", getShardTables.ClusterName, err))
```
Comment (ZuLiangWang) [inline server/service/http/api.go:162]:
```suggestion
		a.respondError(writer, ErrGetCluster, fmt.Sprintf("get cluster failed, clusterName: %s, cause: %s", getShardTables.ClusterName, err.Error()))
```

Comment (ZuLiangWang) [reply server/service/http/api.go:162]: We do not need to exposed error stack to `CeresDB`, just throw error message is enough.

File: server/service/http/api.go [hunk 3]
```diff
@@ -150,15 +150,16 @@ func (a *API) getShardTables(writer http.ResponseWriter, req *http.Request) {
...
+		a.respondError(writer, ErrParseRequest, fmt.Sprintf("decode request body failed, cause: %s", err.Error()))
 		return
 	}
 	log.Info("get shard tables request", zap.String("request", fmt.Sprintf("%+v", getShardTables)))
 
 	c, err := a.clusterManager.GetCluster(req.Context(), getShardTables.ClusterName)
 	if err != nil {
 		log.Error("get cluster failed", zap.String("clusterName", getShardTables.ClusterName), zap.Error(err))
-		a.respondError(writer, ErrGetCluster, fmt.Sprintf("get cluster failed, clusterName:%s", getShardTables.ClusterName))
+		a.respondError(writer, ErrGetCluster, fmt.Sprintf("get cluster failed, clusterName: %s, cause: %s", getShardTables.ClusterName, err.Error()))
```
Comment (ShiKaiWi) [inline server/service/http/api.go:162]: I guess there is no space between the key and value in the log.

**Pull request merged**